src/core/src/pass/constant_folding.cpp

```diff
@@ -96,12 +101,16 @@ bool ov::pass::ConstantFolding::run_on_model(const std::shared_ptr<ov::Model>& m
     return rewritten;
 }
 
-void ov::pass::ConstantFolding::copy_runtime_info_to_target_inputs(const std::shared_ptr<Node>& node,
-                                                                   const Output<Node>& replacement) {
-    for (auto& input : replacement.get_target_inputs()) {
-        auto consumer = input.get_node()->shared_from_this();
-        copy_runtime_info({node, consumer}, consumer);
+void ov::pass::ConstantFolding::copy_runtime_info_from_input_values(const std::shared_ptr<Node>& node) {
+    if (is_type<ov::opset1::ShapeOf>(node) || is_type<ov::opset3::ShapeOf>(node)) {
+        // Don't propogate names of ShapeOf source node since it is not fused itself
+        return;
+    }
+    ov::NodeVector from = {node};
```

Comment:
this looks odd.
why do we need initialize it with {node} and copy runtime info from node to node?
is it possible to replace line 113 with
`if (!from.empty()) 
     copy_runtime_info(from, node)
`

Reply:
> this looks odd. why do we need initialize it with {node} and copy runtime info from node to node? is it possible to replace line 113 with `if (!from.empty()) copy_runtime_info(from, node) `

Actually we need inode itself. We should merge rt info from inputs with rt info of node instead of just replacing. I added comment in code.